Sources/MongoSwift/MongoCollection+Read.swift

```diff
@@ -24,19 +24,23 @@ extension MongoCollection {
     ) throws -> MongoCursor<CollectionType> {
         let opts = try encodeOptions(options: options, session: session)
         let rp = options?.readPreference?._readPreference
+
+        let connection = try resolveConnection(client: self._client, session: session)
+        let cursor: OpaquePointer = self.withMongocCollection(from: connection) { collPtr in
```

Comment:
in the next PR all of this will be removed into an operation and we won't need to call `resolveConnection` anymore as it will happen in the executor. I just left it like this for now to cut down on the size of the diff in this PR.

ditto for aggregate below.